### PR TITLE
Update vswprintf to mussl 1.1.23

### LINKED ISSUE #9305: vswprintf and swprintf produce incorrect result when over 255 wchars
Code:
```
#include <cstdio>
#include <iostream>
#include <string>
#include <stdarg.h>

#define _MAX_CHARS 8096
std::wstring test_string;

int format_test(const wchar_t* szFormat, ...) {
	wchar_t buf[_MAX_CHARS] = { 0 };

	va_list argList;
	va_start(argList, szFormat);
	int ret = vswprintf(&buf[0], _MAX_CHARS - 1, szFormat, argList);
	va_end(argList);
	if (ret >= _MAX_CHARS) {
		ret = _MAX_CHARS - 1;
	}
	if (ret > 0) {
		test_string.assign(&buf[0], ret);
	}
	return ret;
}

int main(int argc, char** argv) {
	auto longString = L"Emscripten is an LLVM-to-JavaScript compiler. It takes LLVM bitcode - which can be generated from C/C++, using llvm-gcc (DragonEgg) or clang, or any other language that can be converted into LLVM - and compiles that into JavaScript, which can be run on the web (or anywhere else JavaScript can run).\nLinks to demos, tutorial, FAQ, etc: https://github.com/emscripten-core/emscripten/wiki\nMain project page: http://emscripten.org\nLicense\nEmscripten is available under 2 licenses, the MIT license and the University of Illinois/NCSA Open Source License.\nBoth are permissive open source licenses, with little if any practical difference between them.\nThe reason for offering both is that (1) the MIT license is well-known, while (2) the University of Illinois/NCSA Open Source License allows Emscripten's code to be integrated upstream into LLVM, which uses that license, should the opportunity arise.\nSee LICENSE for the full content of the licenses.";
	int result = format_test(L"%ls", longString);
	std::wcout << L"result:" << result << L", " << test_string << std::endl;
}
```
Compile like: `emcc test.cpp -std=c++11 -o test.html`

The output string doesn't match the input, this is what gets printed: 
`result:947, Emscripten is an LLVM-to-JavaScript compiler. It takes LLVM bitcode - which can be generated from C/C++, using llvm-gcc (DragonEgg) or clang, or any other language that can be converted into LLVM - and compiles that into JavaScript, which can be run on the Emscripten is an LLVM-to-JavaScript compiler. It takes LLVM bitcode - which can be generated from C/C++, using llvm-gcc (DragonEgg) or clang, or any other language that can be converted into LLVM - and compiles that into JavaScript, which can be run on thewEmscripten is an LLVM-to-JavaScript compiler. It takes LLVM bitcode - which can be generated from C/C++, using llvm-gcc (DragonEgg) or clang, or any other language that can be converted into LLVM - and compiles that into JavaScript, which can be run on theeEmscripten is an LLVM-to-JavaScript compiler. It takes LLVM bitcode - which can be generated from C/C++, using llvm-gcc (DragonEgg) or clang, or any other language that can be `
Not only does it not include the new lines, it's incorrect after 256 chars. All the Emscripten vswprintf tests test small (<256 char) strings, so this didn't get picked up by them.

This was generated by Emscripten version 1.38.37. This is still an issue in 1.38.43.

### PR DESCRIPTION
**Pull request by VirtualTim**
Copy-pasted updated source from http://git.musl-libc.org/cgit/musl/tree/src/stdio/vswprintf.c.
Also added a test case for this.
Fixes #9305.

### CODE REVIEW

File: tests/core/test_wprintf.out [hunk 1]
```diff
@@ -27,6 +27,9 @@ vswoutput st-rts with 0x74
 vsw    continues with 0x65
 vsw    continues with 0x73
 test string has 36 wide characters.
+PrintBigWide wrote 426 wchars:
+test string has 425 wide characters.
```
Comment (VirtualTim) [inline tests/core/test_wprintf.out]: Note: difference in the above is expected, as "%d" is 2 chars, get gets printed as "425", which is 3 chars.

File: tests/core/test_wprintf.cpp [hunk 2]
```diff
@@ -8,16 +8,19 @@
...
-  memset(buffer, 0, 256);
+  wchar_t buffer[MAX_CHARS_SMALL];
+  memset(buffer, 0, MAX_CHARS_SMALL);
   va_list args;
   va_start ( args, format );
   wprintf(L"format    starts with 0x%x\n", *(int*)format);
   wprintf(L"fmt    continues with 0x%x\n", *(((int*)format) + 1));
   wprintf(L"fmt    continues with 0x%x\n", *(((int*)format) + 2));
-  int r = vswprintf ( buffer, 256, format, args );
+  int r = vswprintf ( buffer, MAX_CHARS_SMALL-1, format, args );
```
Comment (VirtualTim) [inline tests/core/test_wprintf.cpp]: Looking at the doco the second arg should be the (number of chars in buffer) -1, so I figured I should change that as well.

**Comment by VirtualTim** [comment]
Looks like `test-upstream-wasm2` failed due to a networking error.

**Comment by VirtualTim** [comment]
Note really sure what sort of thing I should be writing in that readme, but I took a crack at it. Is this the sort of thing you were after?

**Comment by VirtualTim** [comment]
Looks like failures were due to networking issues again.

**Comment by kripken** [comment]
Thanks @VirtualTim, lgtm. Restarted those jobs. Let's also see what @sbc100 thinks of the readme docs.

File: tests/core/test_wprintf.cpp [hunk 3]
```diff
@@ -26,6 +29,22 @@ void PrintWide ( const wchar_t * format, ... )
...
 }
 
+void PrintBigWide ( const wchar_t * format, ... )
+{
+  wchar_t buffer[MAX_CHARS_BIG] = { 0 };
+  va_list args;
+  va_start ( args, format );
+  int ret = vswprintf ( buffer, MAX_CHARS_BIG-1, format, args );
+  va_end ( args );
+  if (ret >= MAX_CHARS_BIG) {
```
Comment (sbc100) [inline tests/core/test_wprintf.cpp]: Can this ever return more than  MAX_CHARS_BIG-1?  It seems not, in which case this condition can be removed?

File: tests/core/test_wprintf.cpp [hunk 4]
```diff
@@ -26,6 +29,22 @@ void PrintWide ( const wchar_t * format, ... )
...
+    ret = MAX_CHARS_BIG - 1;
+  }
+  if (ret > 0) {
+    wprintf(L"PrintBigWide wrote %d wchars:\n", ret);
```
Comment (sbc100) [inline tests/core/test_wprintf.cpp]: Make this unconditional?

**Comment by VirtualTim** [comment]
Thanks for the review sbc100. I've added the suggested changes.

**Comment by VirtualTim** [comment]
@kripken, @sbc100 Anything preventing this being merged?

**Comment by kripken** [comment]
I think merging in latest incoming will fix those errors. Then we can merge it. Sorry for the delay!

File: system/lib/libc/README.md [hunk 5]
```diff
@@ -1,4 +1,5 @@
-This folder contains the musl version of libc at `/musl`.  The upstream version can be found at http://www.musl-libc.org/
+This folder contains the musl version of libc at `/musl`. The upstream version can be found at http://www.musl-libc.org/.
+Most of the source comes from musl 1.1.15. Some is from older versions, but I'm assuming that this is because these files haven't changed between versions.
```
Comment (sbc100) [inline system/lib/libc/README.md]: Can you drop this last sentence of re-phrase.  The last update was indeed to 1.1.15 in #4813.

Comment (VirtualTim) [reply system/lib/libc/README.md]: Done.

**Comment by VirtualTim** [comment]
Merged in the latest incoming, and now SIMD tests are failing.

**Comment by kripken** [comment]
Odd error - I saw it myself last night too. But it's not happening on incoming, and can't be related to this PR. Seems ok to land this. Thanks!

**Pull request merged**